components/eamxx/src/share/io/scream_io_control.hpp

```diff
@@ -22,23 +22,37 @@ struct IOControl {
   util::TimeStamp next_write_ts;
   util::TimeStamp last_write_ts;
 
+  // At run time, set dt in the struct, so we can compute next_write_ts correctly,
+  // even if freq_units is "nsteps"
+  // NOTE: this ASSUMES dt is constant throughout the simulation. Error is thrown if it changes
+  double dt = 0;
```

Comment:
Well, it is necessary. When we write, we need to compute the next write timestamp, to determine if the next snap will fit in the current file. If dt changes, then we cannot close a file if full, since we can't tell if the next snap will fit. More precisely:
- for storage type "NumSnaps", then we don't really need to know `next_write_ts` to tell if it will fit
- for storage type "one_month"/"one_year" (and maybe in the future "one_day"), we need to be able to compute `next_write_ts` (to check the date). If the output freq is given in nsteps units, then we compute it as `last_write_ts + dt*freq`.

We could argue that, since dt_atm will always divide 86400, increasing/decreasing it at runtime will not affect whether the next snap will have the needed date or not (to see if it fits).

Reply:
That is allowed. We only request that once you set dt during the run, you don't change it. That is, no time adaptivity (which will likely never happen anyways, but I added the check just in case).

Reply:
That's fair. I can make it clearer